Makefile

```diff
@@ -84,41 +80,57 @@ TEMP_DIR := $(shell mktemp -d)
 all: all-container
 
 sub-container-%:
-	$(MAKE) --no-print-directory ARCH=$* container
+	# 'clean' is necessary since build process needs to overwrite kube-state-metrics binary. Otherwise all images end up with binary made for one architecture
+	$(MAKE) --no-print-directory ARCH=$* clean .container-$*
 
 sub-push-%:
-	$(MAKE) --no-print-directory ARCH=$* push
+	# 'clean' is necessary since build process needs to overwrite kube-state-metrics binary. Otherwise all images end up with binary made for one architecture
+	$(MAKE) --no-print-directory ARCH=$* clean .quay-push-$*
 
 all-container: $(addprefix sub-container-,$(ALL_ARCH))
 
 all-push: $(addprefix sub-push-,$(ALL_ARCH))
 
-container: .container-$(ARCH)
-.container-$(ARCH): kube-state-metrics
-	cp -r * "${TEMP_DIR}"
-	${DOCKER_CLI} build -t $(MULTI_ARCH_IMG):$(TAG) "${TEMP_DIR}"
-	${DOCKER_CLI} tag $(MULTI_ARCH_IMG):$(TAG) $(MULTI_ARCH_IMG):latest
-	rm -rf "${TEMP_DIR}"
+manifest-tool:
+	curl -fsSL https://github.com/estesp/manifest-tool/releases/download/v1.0.2/manifest-tool-linux-amd64 > ./manifest-tool
```

Comment:
Let's move the `manifest-tool` version to a variable

Reply:
Let's also have a bin existence check, so that we don't run this every time. Something like this:

HAS_GOLANGCI := $(shell which golangci-lint) which is used in the `lint` make target

Reply:
Simple checking for binary existence is a contrast to requiring a version of a binary. Consider having a pre-existing system-wide binary in version A, but specifying version B in Makefile. In such scenario you would need to add another check to validate globally installed binary version against version requested in Makefile. It is a question of code simplicity vs first execution speed.